gorilla/gorilla.go

```diff
@@ -43,12 +44,85 @@ func ActionName(route *mux.Route, actionName string) {
 	})
 }
 
+var allMethods = []string{"GET", "POST", "PUT", "DELETE", "PATCH", "HEAD", "OPTIONS"}
+
+type hi struct {
+	methods map[string]bool
+	handler handler
+}
+
+func (i *hi) addMethods(methods []string) *hi {
+	for _, m := range methods {
+		i.methods[m] = true
+	}
+	return i
+}
+
+func (i *hi) methodComplement() []string {
+	c := []string{}
+	for _, m := range allMethods {
+		if !i.methods[m] {
+			c = append(c, m)
+		}
+	}
+	return c
+}
+
+// addMethodNotAllowedHandlers installs complementary handlers to the given router, for
+// all logjam defined routes.
+func addMethodNotAllowedHandlers(router *mux.Router) {
+	handlers := map[string]*hi{}
+	if router.MethodNotAllowedHandler == nil {
+		router.MethodNotAllowedHandler = http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
This needs to set the logjam action. Otherwise it's just what gorilla/mux is doing already.

Reply:
Setting the the action name is not necessary, since router.MethodNotAllowedHandler is never called by gorilla, it is called by the route that wraps it. See https://github.com/xing/logjam-agent-go/blob/add-method-not-allow-handlers/gorilla/gorilla_test.go#L106.